File: chroma-league/src/test/java/net/booone/chromaleague/rest/http/handlers/ActivePlayerResponseHandlerTest.java
package net.booone.chromaleague.rest.http.handlers;

import net.booone.chromaleague.rest.activeplayer.ActivePlayer;
import net.booone.chromaleague.rest.activeplayer.ChampionStats;
import net.booone.chromaleague.rest.http.LeagueHttpClientMock;
import org.apache.http.HttpResponse;
import org.junit.jupiter.api.Test;

import java.io.IOException;
import java.net.URISyntaxException;
import java.util.Optional;

import static org.junit.jupiter.api.Assertions.*;

public class ActivePlayerResponseHandlerTest {
    @Test
    void testActivePlayerParsing() throws IOException, URISyntaxException {
        //given
        HttpResponse testResponseFromJSON = LeagueHttpClientMock.createTestResponseFromJSON("json/activeplayer.json");

        //when
        Optional<ActivePlayer> activePlayer = new ActivePlayerResponseHandler().handleResponse(testResponseFromJSON);

        //then
        assertTrue(activePlayer.isPresent());
        verifyActivePlayer(activePlayer.get());
    }

    public static void verifyActivePlayer(ActivePlayer activePlayer) {
        assertNotNull(activePlayer);
        assertEquals(123.45, activePlayer.currentGold());
        assertEquals(1, activePlayer.level());

        final ChampionStats championStats = activePlayer.championStats();
        assertEquals(255.59997, championStats.currentHealth());
        assertEquals(655.5999755859375, championStats.maxHealth());
        assertEquals(111.11111111111111, championStats.resourceValue());
        assertEquals(222.222222222222222, championStats.resourceMax());
        assertEquals("MANA", championStats.resourceType());
    }
}

File: test/rest/http/handlers/ActivePlayerResponseHandler.test.ts
import {describe, expect, test} from 'vitest';
import type {ActivePlayer} from '../../../../src/rest/activeplayer/ActivePlayer.js';

describe('ActivePlayerResponseHandler', () => {
    test('testActivePlayerParsing', () => {
        //given
        const json = {
            championStats: {
                attackRange: 125.0,
                currentHealth: 255.59997,
                maxHealth: 655.5999755859375,
                resourceValue: 111.11111111111111,
                resourceMax: 222.222222222222222,
                resourceType: 'MANA',
            },
            currentGold: 123.45,
            level: 1,
            riotId: 'BonE#EUPL',
            riotIdGameName: 'BonE',
        };

        //when
        const activePlayer = json as ActivePlayer;

        //then
        expect(activePlayer).toBeDefined();
        verifyActivePlayer(activePlayer);
    });
});

export function verifyActivePlayer(activePlayer: ActivePlayer): void {
    expect(activePlayer).toBeDefined();
    expect(activePlayer.currentGold).toBe(123.45);
    expect(activePlayer.level).toBe(1);

    const championStats = activePlayer.championStats;
    expect(championStats.currentHealth).toBe(255.59997);
    expect(championStats.maxHealth).toBe(655.5999755859375);
    expect(championStats.resourceValue).toBe(111.11111111111111);
    expect(championStats.resourceMax).toBe(222.222222222222222);
    expect(championStats.resourceType).toBe('MANA');
}
